plugin/keyring_vault/vault_credentials.cc

```diff
@@ -0,0 +1,18 @@
+#include "vault_credentials.h"
+
+namespace keyring
+{
+
+const Secure_string& get_credential(const Vault_credentials &credentials, const Secure_string &key)
+{
+  static Secure_string empty_value;
```

Comment:
AFAIK, empty_value is not thread-safe in c++03 - that it can get initialized multiple times. This is not a problem here if it gets initialized more than once. Also, this function is called only on plugin initialization. I find it better this way than polluting keyring namespace.